packages/flutter/test/material/scaffold_test.dart

```diff
@@ -446,6 +446,139 @@ void main() {
+
+        await tester.pump();
+
+        final Offset snackBarBottomRight = tester.getBottomRight(find.byType(SnackBar));
+        final Offset bottomNavigationBarTopRight = tester.getTopRight(find.byKey(boxKey));
+
+        expect(snackBarBottomRight.dy, equals(bottomNavigationBarTopRight.dy));
+      });
+    }
+
+    testWidgets('behavior is fixed and scaffold has floating action button', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(home: Scaffold(
+          body: Container(),
+          floatingActionButton: FloatingActionButton(onPressed: () {},),
+        ),
+        ),
+      );
+
+      final ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+      scaffoldState.showSnackBar(const SnackBar(content: Text('Snackbar text'), behavior: SnackBarBehavior.fixed,));
+
+      await tester.pump();
+
+      final Offset snackBarBottomRight = tester.getBottomRight(find.byType(SnackBar));
+      final Offset scaffoldBottomRight = tester.getBottomRight(find.byType(Scaffold));
+
+      expect(snackBarBottomRight.dy, equals(scaffoldBottomRight.dy));
+    });
+
+    testWidgets('behavior is floating and scaffold has floating action button', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(home: Scaffold(
+          body: Container(),
+          floatingActionButton: FloatingActionButton(onPressed: () {},),
+        ),
+        ),
+      );
+
+      final ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+      scaffoldState.showSnackBar(const SnackBar(content: Text('Snackbar text'), behavior: SnackBarBehavior.floating,));
+
+      await tester.pump();
+
+      final Offset snackBarBottomRight = tester.getBottomRight(find.byType(SnackBar));
+      final Offset fabTopRight = tester.getTopRight(find.byType(FloatingActionButton));
+
+      expect(snackBarBottomRight.dy, equals(fabTopRight.dy));
+    });
+
+    testWidgets('behavior is fixed and scaffold has FAB and bottom navigation bar', (WidgetTester tester) async {
+      final UniqueKey boxKey = UniqueKey();
+      await tester.pumpWidget(
+        MaterialApp(home: Scaffold(
+          body: Container(),
+          bottomNavigationBar: SizedBox(key: boxKey, width: 800, height: 60),
+          floatingActionButton: FloatingActionButton(onPressed: () {},),
+        ),
+        ),
+      );
+
+      final ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+      scaffoldState.showSnackBar(const SnackBar(content: Text('Snackbar text'), behavior: SnackBarBehavior.fixed,));
+
+      await tester.pump();
+
+      final Offset snackBarBottomRight = tester.getBottomRight(find.byType(SnackBar));
+      final Offset bottomNavigationBarTopRight = tester.getTopRight(find.byKey(boxKey));
+
+      expect(snackBarBottomRight.dy, equals(bottomNavigationBarTopRight.dy));
+    });
+
+    testWidgets('behavior is floating and scaffold has FAB and bottom navigation bar', (WidgetTester tester) async {
```

Comment:
Same comments as above regarding the title of the test and code indentation/style.

Reply:
Fixed.